tgui/packages/tgui/interfaces/GasCanister/DetonatorTimer.js

```diff
@@ -0,0 +1,50 @@
+import { Box, AnimatedNumber } from '../../components';
+
+export const DetonatorTimer = props => {
+  const {
+    time,
+    warningThreshold = 30,
+    dangerThreshold = 10,
+    explosionMessage = "BO:OM",
+  } = props;
+
+  const FormatTime = () => {
+    let seconds = Math.floor(time % 60);
+    let minutes = Math.floor((time - seconds) / 60);
+    if (time <= 0) {
+      return explosionMessage;
+    }
+    if (seconds < 10) {
+      seconds = `0${seconds}`;
+    }
+    if (minutes < 10) {
+      minutes = `0${minutes}`;
+    }
+
+    return `${minutes}:${seconds}`;
+  };
```

Comment:
This would be better brought out into a thing in common/ imo

Reply:
`tgui/format.js`

Reply:
moved into `tgui/format.js`